src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -56,31 +58,27 @@ public class LuceneFieldMapper extends KNNVectorFieldMapper {
         );
 
         this.knnMethod = input.getKnnMethodContext();
-
-        this.fieldType = new FieldType();
-
-        this.fieldType.setTokenized(false);
-        this.fieldType.setIndexOptions(IndexOptions.NONE);
-
         final SpaceType spaceType = this.knnMethod.getSpaceType();
         final VectorSimilarityFunction vectorSimilarityFunction = Optional.ofNullable(
             SPACE_TYPE_TO_VECTOR_SIMILARITY_FUNCTION.get(spaceType)
         ).orElseThrow(() -> new IllegalArgumentException(String.format("Space type [%s] is not supported for Lucene engine", spaceType)));
-
         final int dimension = input.getMappedFieldType().getDimension();
-        if (dimension > MAX_DIMENSION) {
-            throw new IllegalArgumentException(
-                String.format(
-                    "Dimension value cannot be greater than [%s] but got [%s] for vector [%s]",
-                    MAX_DIMENSION,
-                    dimension,
-                    input.getName()
-                )
-            );
+
+        this.fieldType = KnnVectorField.createFieldType(dimension, vectorSimilarityFunction);
+
+        if (this.hasDocValues) {
+            this.docValuesFieldType = buildDocValuesFieldType(this.knnMethod.getKnnEngine());
```

Comment:
[Question] why do we need this variable, how this is getting used? Cannot see in this PR.

Reply:
> The build is also failing, any reason why this is failing?

Spotless check I forgot to run

Reply:
This value provides information for the VectorField. It is used in VectorField construction.

Reply:
Got. I missed the parameter that you are passing to the create the doc value.